src/hotspot/share/opto/type.hpp

```diff
@@ -1682,13 +1696,22 @@ inline const TypeAry *Type::isa_ary() const {
   return ((_base == Array) ? (TypeAry*)this : NULL);
 }
 
+inline const TypeVectMask *Type::is_vectmask() const {
+  assert( _base == VectorM, "Not a Vector Mask" );
+  return (TypeVectMask*)this;
+}
+
+inline const TypeVectMask *Type::isa_vectmask() const {
+  return (_base == VectorM) ? (TypeVectMask*)this : NULL;
+}
+
 inline const TypeVect *Type::is_vect() const {
-  assert( _base >= VectorA && _base <= VectorZ, "Not a Vector" );
+  assert( _base >= VectorM && _base <= VectorZ, "Not a Vector" );
```

Comment:
For is_vect() can we make it only valid for vector type while invalid for mask type? I think that sounds more reasonable, and could lower risk, given that there are a lot of is_vect() usages in existing code.

Reply:
Fixed this, thanks!

Reply:
Hi @nsjian , @iwanowww, 

On a second thought, we may enforce strict base type constraint (base >= VectorA && < base <= VectorZ)  on TypeVect::isa_vect routine, but  TypeVect::is_vect should be relaxed to include TypeVectorMask , considering TypeVectMask is a subclass of TypeVect and thus qualifies to be a vector type.  It will simplify adding conditional checks across the code. 
Since X86 supports two sorts of masks types (a TypeVect for non-AVX512 and a TypeVectMask for AVX-512) thus relaxing the constraint on TypeVect::is_vect will save un-necessary complexity.  

Looking forward to your feedback on this. 

Best Regards